apps/src/code-studio/components/ShareAllowedDialog.story.jsx

```diff
@@ -123,3 +126,11 @@ WithSharingForUserDisabled.args = {
   isUnpublishPending: true,
   userSharingDisabled: true
 };
+
+export const InRestrictedShareMode = Template.bind({});
+InRestrictedShareMode.args = {
+  ...defaultArgs,
+  canPublish: true,
+  inRestrictedShareMode: true,
+  appType: 'spritelab'
```

Comment:
`appType` is `poetry` in Poetry. I think everything still works because we are basing `shouldRestrict off of the app being [blockly](https://github.com/code-dot-org/code-dot-org/pull/50269/files#diff-d86b69ce994e9c19fad1becf664093b55fcb57f48e5b3f842c2039aba47368ccR114)
'